types/contract-proxy-kit/index.d.ts

```diff
@@ -67,3 +67,44 @@ declare class CPK {
         options?: CPK.TransactionOptions,
     ): Promise<CPK.TransactionResult>;
 }
+
+declare class CPKProvider {
+    static getContractAddress(contract: any): string;
+    static attemptTransaction(
+        contract: any,
+        viewContract: any,
+        methodName: string,
+        params: ReadonlyArray<object>,
+        sendOptions: CPK.TransactionOptions,
+        err: any
+    ): Promise<CPK.TransactionResult>;
+    static getSendOptions(options: CPK.TransactionOptions, ownerAccount?: string): object;
+    
+    init(
+        isConnectedToSafe: boolean,
+        ownerAccount: string,
+        masterCopyAddress: string,
+        proxyFactoryAddress: string,
+        multiSendAddress: string
+    ): object;
+    getProvider(): any;
+    getNetworkId(): Promise<number>;
+    getOwnerAccount(): Promise<string>;
+    getCodeAtAddress(contract: any): Promise<string>;
+    getContract(abi: ReadonlyArray<object>, address: string): any;
+    checkSingleCall(tx: CPK.Transaction): Promise<any>;
+    attemptSafeProviderSendTx(
+        tx: CPK.Transaction,
+        options: CPK.TransactionOptions
+    ): Promise<object>;
+    attemptSafeProviderMultiSendTxs(transactions: ReadonlyArray<CPK.Transaction>): object;
+    encodeMultiSendCallData(transactions: ReadonlyArray<CPK.Transaction>): string;
+}
```

Comment:
The cpkProviders have now 3 new methods:
```
encodeAttemptTransaction(contractAbi: ReadonlyArray<object>, methodName: string, params: ReadonlyArray<object>): string;
getGasPrice(): Promise<number>;
getSafeNonce(safeAddress: string): Promise<number>;
```

Reply:
@cag could you add them? The last time I tried to push my commits from my forked repo to yours I couldn't find this branch.

Reply:
This methods were added here: https://github.com/gnosis/contract-proxy-kit/pull/49

Reply:
I just added them :)